src/components/web-monitoring-ui.jsx

```diff
@@ -78,20 +87,56 @@ export default class WebMonitoringUi extends React.Component {
       });
   }
 
+  loadUser () {
+    api.isLoggedIn()
+      .then(loggedIn => {
+        this.setState({user: api.userData, loading: false});
+      });
+  }
+
   componentWillMount () {
-    this.loadPages();
+    this.loadUser();
+    this.setState({loading: true});
   }
 
   render () {
-    const withData = bindComponent({pages: this.state.pages, user: this.state.user});
+    // TODO: There's probably a way to compose these objects to cut down on repetition
+    const withData = bindComponent({
+      pages: this.state.pages,
+      user: this.state.user,
+    });
+    const withDataAll = bindComponent({
+      pages: this.state.pages,
+      user: this.state.user,
+      loadPages: this.loadPages,
+      showAll: true,
+    });
+    const withDataMyDomains = bindComponent({
+      pages: this.state.pages,
+      user: this.state.user,
+      loadPages: this.loadPages,
+      showAll: false,
+    });
```

Comment:
This was a convenient tool when we applied the same data to every view, but that’s no longer the case, so I think we should just ditch `bindComponent` here. Instead, we should just have the `render` prop for each route do the work, e.g:

```jsx
<Route path="/all" render={routeProps => {
  // this check should actually probably be in `loadPages`, maybe with something that checks
  // for stale-ness, e.g. "load again if it's been more than 10 minutes"
  if (!this.state.pages) {
    this.loadPages(true);
  }
  return <PageList ...routeProps pages={this.state.pages} user={this.state.user} />
}} />
```

Or a little fancier:

```jsx
const withData = (ComponentType, pageType) => {
  return (routeProps) => {
    const pages = this.state[pageType];
    if (!pages) {
      this.loadPages(pageType === 'pages');
    }
    return <ComponentType ...routeProps pages={pages} user={this.state.user} />;
  };
};

// ...all the existing stuff in `render()`...

<Route path="/all" render={withData(PageList, 'pages')} />
<Route path="/mydomains" render={withData(PageList, 'assignedPages')} />
```

There are probably a lot of other ways to wrap up this functionality if you think through it (e.g. container components in Redux). The main point is that I don’t think `bindComponent` is serving our needs well here anymore.